src/main/kotlin/racingcar/interfaces/InputAdapter.kt

```diff
@@ -0,0 +1,63 @@
+package racingcar.interfaces
+
+import racingcar.domain.Car
+
+class InputAdapter(
+  private val carNames: String,
+  private val matches: String
+) {
+  fun splitCarName(): List<String> {
+    val cars = carNames.split(DELIMITER)
+    return cars
+  }
+
+  fun validateCarName(cars: List<String>): Set<String> {
+    val racingCars = mutableSetOf<String>()
+    for (car in cars) {
+      when {
+        car.length > 5 -> throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_LESS_THAN_5_EXCEPTION_MSG)
+        car.isEmpty() -> throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_EMPTY_EXCEPTION_MSG)
+        car.isBlank() -> throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_WHITESPACE_EXCEPTION_MSG)
+        racingCars.contains(car) -> throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_SAME_CARNAME_EXCEPTION_MSG)
+      }
+      racingCars.add(car)
+    }
+    if (racingCars.size == 1) throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_ONE_EXCEPTION_MSG)
+    return racingCars
+  }
+
+  fun changeTypeOfCar(racingCars: Set<String>): List<Car> {
+    val readyRacingCars: List<Car> = racingCars.map { Car(it) }
+    return readyRacingCars
+  }
+
+  fun changeTypeOfMatches(): Int {
+    validateMatches()
+    return matches.toInt()
+  }
+
+  fun validateMatches(): Boolean {
+    when {
+      matches.toIntOrNull() == null -> throw IllegalArgumentException(INPUT_ERROR_MSG + MATCHES_NOT_INT_EXCEPTION_MSG)
+      matches.toInt() == 0 -> throw IllegalArgumentException(INPUT_ERROR_MSG + MATCHES_ZERO_EXCEPTION_MSG)
+      matches.toInt() < 0 -> throw IllegalArgumentException(INPUT_ERROR_MSG + MATCHES_NEGATIVE_NUMBER_EXCEPTION_MSG)
+    }
+    return true
```

Comment:
이 함수에서 반환하는 값이 true인지 false인지가 코드에서는 중요하지 않은것 같아요
이렇게 변경하면 조금이라도 다이어트된 코드가 될것 같아요
```suggestion
    fun validateMatches() {
    when {
      // 조건 1
      // 조건 2
      // 조건 3
    }
  }
```

Reply:
```
  fun changeTypeOfMatches(): Int {
    validateMatches()
    return matches.toInt()
  }
```

validateMatches가 true이면 숫자로 변환해줍니다!